include/uart.h

```diff
@@ -245,16 +437,157 @@ struct uart_driver_api {
+ *
+ * @param dev UART device structure.
+ *
+ * @retval -EFAULT There is no active transmission.
+ * @retval 0	   If successful, negative errno code otherwise.
+ */
+int uart_tx_abort(struct device *dev)
+{
+	const struct uart_driver_api *api =
+			(const struct uart_driver_api *)dev->driver_api;
+
+	return api->tx_abort(dev);
+}
+
+/**
+ * @brief Start receiving data through UART.
+ *
+ * Function sets given buffer as first buffer for receiving and returns
+ * immediately. After that event handler, set using @ref uart_set_callback,
+ * is called with UART_RX_RDY or UART_RX_BUF_REQUEST events.
+ *
+ * @param dev UART device structure.
+ * @param buf Pointer to receive buffer.
+ * @param len Buffer length.
+ *
+ * @retval -EBUSY RX already in progress.
+ * @retval 0	  If successful, negative errno code otherwise.
+ *
+ */
+int uart_rx_enable(struct device *dev, u8_t *buf, size_t len)
+{
+	const struct uart_driver_api *api =
+				(const struct uart_driver_api *)dev->driver_api;
+
+	return api->rx_enable(dev, buf, len);
+}
+
+/**
+ * @brief Provide receive buffer in response to UART_RX_BUF_REQUEST event.
+ *
+ * Provide pointer to RX buffer, which will be used when current buffer is
+ * filled.
+ *
+ * @param dev UART device structure.
+ * @param buf Pointer to receive buffer.
+ * @param len Buffer length.
+ *
+ * @retval -EBUSY Next buffer already set.
+ * @retval 0	  If successful, negative errno code otherwise.
+ *
+ */
+int uart_rx_buf_rsp(struct device *dev, u8_t *buf, size_t len)
+{
+	const struct uart_driver_api *api =
+				(const struct uart_driver_api *)dev->driver_api;
+
+	return api->rx_buf_rsp(dev, buf, len);
+}
+
+/**
+ * @brief Set RX timeout function.
+ *
+ * @param dev     UART device structure.
+ * @param timeout Timeout
+ *
+ * @retval 0 If successful, negative errno code otherwise.
+ */
+int uart_rx_timeout_set(struct device *dev, u32_t timeout)
+{
+	const struct uart_driver_api *api =
+			(const struct uart_driver_api *)dev->driver_api;
+
+	return api->rx_timeout_set(dev, timeout);
+}
+
+/**
+ * @brief Disable RX
+ *
+ * UART_RX_BUF_RELEASED event will be generated for every buffer scheduled,
+ * after that UART_RX_DISABLED event will be generated.
+ *
+ * @param dev UART device structure.
+ *
+ * @retval -EFAULT There is no active reception.
+ * @retval 0	   If successful, negative errno code otherwise.
+ */
+void uart_rx_disable(struct device *dev)
+{
+	const struct uart_driver_api *api =
+			(const struct uart_driver_api *)dev->driver_api;
+
+	return api->rx_disable(dev);
+}
+
+#endif
+
 /**
  * @brief Check whether an error was detected.
  *
  * @param dev UART device structure.
  *
- * @retval UART_ERROR_OVERRUN if an overrun error was detected.
- * @retval UART_ERROR_PARITY if a parity error was detected.
- * @retval UART_ERROR_FRAMING if a framing error was detected.
- * @retval UART_ERROR_BREAK if a break error was detected.
- * @retval 0 Otherwise.
+ * @retval uart_rx_stop_reason If error during receiving occurred.
```

Comment:
This enumeration still has non-negative values: `UART_ERROR_OVERRUN` will be undistinguishable from successful return.